R/rename.r

```diff
@@ -12,11 +14,20 @@
 #' @examples
 #' x <- c("a" = 1, "b" = 2, d = 3, 4)
 #' # Rename column d to "c", updating the variable "x" with the result
-#' x <- rename(x, replace=c("d" = "c"))
+#' x <- rename(x, replace = c("d" = "c"))
 #' x
 #' # Rename column "disp" to "displacement"
 #' rename(mtcars, c("disp" = "displacement"))
-rename <- function(x, replace, warn_missing = TRUE) {
+rename <- function(x, replace, warn_missing = TRUE, warn_duplicate = TRUE ) {
+
+  # This line does the real work of `rename()`.
   names(x) <- revalue(names(x), replace, warn_missing = warn_missing)
+
+  # Check if any names are duplicated.
+  duplicated_names <- names(x)[duplicated(names(x))]
+  if (warn_duplicate && (length(duplicated_names) > 0L)) {
+    duplicated_names_message <- paste(duplicated_names, collapse="`, `")
+    warning("The plyr::rename operation has created duplicates for the following name(s): (`", duplicated_names_message, "`)")
```

Comment:
Can you wrap those lines please? And all `call. = FALSE`.  

I think it's slightly more elegant to do `paste("`", duplicated_names, "`", collapse = ", ")`